pkg/apis/apps/v1alpha1/apimanager_types.go

```diff
@@ -229,9 +229,14 @@ type SystemPVCSpec struct {
 }
 
 type SystemS3Spec struct {
-	AWSBucket      string                  `json:"awsBucket"`
-	AWSRegion      string                  `json:"awsRegion"`
-	AWSCredentials v1.LocalObjectReference `json:"awsCredentialsSecret"`
+	// TODO we now have a problem which is that some fields that were required
+	// when specifying S3 as the File Storage should no longer be required as
+	// the configuration is located in ConfigurationSecretName. What happens
+	// with existing installations?
+	AWSBucket              string                  `json:"awsBucket"`
```

Comment:
This is what concerns me more about all:
We had some required fields in the CR to provide the S3 configuration that we have decided we are not going to use anymore, but they are required. And also, we are adding a new required field which is the configuration secret.
I think this breaks compatibility right? because the existing deploymets do not have the new ConfigurationSecretRef required parameter.

Reply:
One proposal:

* deprecate `spec.filestorage.amazonSimpleStorageService`which is optional
* create new optional field: `spec.filestorage.s3secret` with `v1.LocalObjectReference` type.

Upgrading path:

Gather all the information from `spec.filestorage.amazonSimpleStorageService` and `aws-auth` s3 secret and update `aws-auth` s3 secret with all data. Update CR to look like something like this:

```
spec:
  system:                                            
    fileStorage:                                     
      s3secret:                    
        name: aws-auth                  
```

Reply:
What you say about deprecating that field and then migrating it to the new configuration secret seems a good solution.
The idea of that new secret is that it HAS to be provided by the user with any name we want. What do we do in the case of the migration that we are the ones creating it? Do we hardcode a specific name and document in the release notes that it is created with that name? For new installations that secret name will not be created as users will use the name  they want.

Regarding the proposed field name, you cannot use numbers as part of the name of the fields because it fails openapi validation related code https://github.com/operator-framework/operator-sdk/issues/1120. That's the reason I chose amazonSimpleStorageService instead of s3 for that section so we'd need to choose another name.

Do you think it might be better to set the s3secret field (required!) inside a new section in fileStorage(amazonsthree for example) instead of being directly a field in the FileStorage section? Keep in mind also that inside the fileStorage section there are also other fields NOT related to S3 so if in the future we need to add more s3 related fields and we directly place the s3secret there we would end up with more than one s3 related field in that section mixed with other filestorage configurability elements, not related to s3, making us not able/really hard to check what deployment method is used for fileStorage. Creating a new specific section for it would allow us to add more optional fields related to s3 in case we need them

Example:
```
spec:
  system:                                            
    fileStorage:                                     
      amazonsthree(this is not going to be the final name probably):                    
        sthreesecret:
          name: aws-auth
```

instead of
```
spec:
  system:                                            
    fileStorage:                                     
      s3secret:                    
        name: aws-auth
```

What do you think?

Reply:
Ok, it allows some extensibility. But I cannot think of anything being outside the secret.